modules/afsocket/socket-options-inet.c

```diff
@@ -39,6 +40,22 @@
 #define SOL_TCP IPPROTO_TCP
 #endif
 
+#ifdef SO_BINDTODEVICE
+static int
+socket_options_inet_set_interface(SocketOptionsInet *self, gint fd)
+{
+  cap_t saved_caps;
+  int result;
+
+  saved_caps = g_process_cap_save();
+  g_process_enable_cap("cap_net_raw");
+  result = setsockopt(fd, SOL_SOCKET, SO_BINDTODEVICE, self->interface_name,
```

Comment:
Optional:
This is kinda hard to read for me with the `?:` operator in the end.
Would you consider chaning like:
```c
result = setsockopt(...);
result = result < 0 ? errno : 0;
...
return result;
```
well maybe this suggestion is not the best, but I would still not *hide* the `?:` operator.